### PR TITLE
fix a bug in service test parameter type validation

### PR DESCRIPTION
**Pull request by YannanGao-gs**
#### What type of PR is this?

Bug fix
#### What does this PR do / why is it needed ?

fix a bug in service test parameter type validation
#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-language-pure-dsl-service/src/main/java/org/finos/legend/engine/language/pure/dsl/service/compiler/toPureGraph/HelperServiceBuilder.java [hunk 1]
```diff
@@ -290,4 +291,21 @@ else if (params instanceof MultiExecutionParameters)
         }
         throw new UnsupportedOperationException("Unsupported service execution type '" + params.getClass().getSimpleName() + "'");
     }
+
+    private static boolean isTypeCompatibleForAssignment(String type, String assignmentType)
```
Comment (MauricioUyaguari) [inline legend-engine-language-pure-dsl-service/src/main/java/org/finos/legend/engine/language/pure/dsl/service/compiler/toPureGraph/HelperServiceBuilder.java:295]: this method doesn't seem specific to here. Don't we have another method that would check that an instance value is compatible with a type ?

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

**Comment by hardikmaheshwari** [comment]
Lets please add a test case indicating what we are trying to fix. Thanks

**Comment by MauricioUyaguari** [comment]
> Lets please add a test case indicating what we are trying to fix. Thanks

@YannanGao-gs  please add a compile grammar test for what this change fixes

File: legend-engine-language-pure-dsl-service/src/main/java/org/finos/legend/engine/language/pure/dsl/service/compiler/toPureGraph/HelperServiceBuilder.java [hunk 2]
```diff
@@ -178,17 +178,12 @@ public static void validateServiceTestParameterValues(List<Root_meta_legend_serv
...
-                String paramType = param._genericType()._rawType()._name();
-
                 org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.multiplicity.Multiplicity paramValueMultiplicity = paramValue._multiplicity();
-                String paramValueType = paramValue._genericType()._rawType()._name();
-
-                if (!("Nil".equals(paramValueType) || paramType.equals(paramValueType)) || !org.finos.legend.pure.m3.navigation.multiplicity.Multiplicity.subsumes(paramMultiplicity, paramValueMultiplicity))
-                {
-                    throw new EngineException("Parameter value type does not match with parameter type for parameter: '" + param._name() + "'", sourceInformation, EngineErrorType.COMPILATION);
-                }
+               if (!"Nil".equals(paramValue._genericType()._rawType()))
```
[the remark this answers is not in the document]
Comment (hardikmaheshwari) [reply legend-engine-language-pure-dsl-service/src/main/java/org/finos/legend/engine/language/pure/dsl/service/compiler/toPureGraph/HelperServiceBuilder.java:183]: Yes it is needed actually, I can explain offline if needed

**Pull request merged**